### PR TITLE
emit parameter names for static forwarders

### PR DESCRIPTION
**Pull request by ghik**
Fixes https://github.com/scala/bug/issues/10650

### CODE REVIEW

**Comment by ghik** [comment]
Test failures seem to be unrelated to my changes, they fail on `2.12.x` as well when I run them on my local machine.

**Comment by SethTisue** [comment]
test failures reported at https://github.com/scala/scala-dev/issues/457

File: test/files/run/t10650b/Test_2.scala [hunk 1]
```diff
@@ -0,0 +1,13 @@
+object Test extends App {
```
Comment (lrytz) [inline test/files/run/t10650b/Test_2.scala]: I don't see any additional value in also testing this under separate compilation - maybe I'm missing something?

Comment (ghik) [reply test/files/run/t10650b/Test_2.scala]: You may be right. I simply wrote tests similar to the ones for https://github.com/scala/bug/issues/9437

Comment (ghik) [reply test/files/run/t10650b/Test_2.scala]: So should I remove this test? If so, then it probably also makes sense to remove `t9437b`.

Comment (lrytz) [reply test/files/run/t10650b/Test_2.scala]: Ah I see. I don't see the value in separate compilation in the existing tests either. When using a named paramter (``a = 1, `_` = "2"`` etc) in Scala code on a Scala-defined method, the Scala compiler doesn't look at the classfile attribute holding the parameter names. The information is obtained from the "pickle", the serialized chunk of the symbol table. So I'd say let's remove the additional test in this PR.

Comment (lrytz) [reply test/files/run/t10650b/Test_2.scala]: Yes, let's remove those tests.

**Comment by lrytz** [comment]
You can squash all in one commit, and also rebase on current 2.12.x to fix the build failure.

**Comment by ghik** [comment]
All seems good.

**Review by lrytz — APPROVED** [review]
Thanks!

**Pull request merged**